### ISSUE TITLE
Variation Selectors are not considered properly

### ISSUE THREAD

**Issue by ticky** [Ranks/emojione#415]
Several emoji which on most platforms’ native Emoji implementations use variation selectors to decide if the preceding character should render as emoji or simply a normal glyph are erroneously _always_ treated as emoji by EmojiOne.

Examples include;
- `®` (U+00AE), which should be a normal glyph unless followed by U+FE0E VARIATION SELECTOR-15, resulting in `®️`
- `™` (U+2122), which should behave in the same way, with U+2122-U+FE0E resulting in `™️`
- `⚠` (U+26A0) should behave the same way; U+26A0-U+FE0E should result in `⚠️`

**Comment by JoshyPHP** [comment]
For reference, here is the [relevant section about presentation style in UTR-51](http://unicode.org/reports/tr51/index.html#Presentation_Style). For anyone who's not familiar with it, here's a short summary:

 * There are two [variation selectors](https://en.wikipedia.org/wiki/Variation_Selectors_(Unicode_block)) VS15 and VS16 that indicate respectively that a glyph should be displayed as text or as an image.
 * Without a selector, the decision is left to the browser/application. The Unicode guidelines are:
    * Word processors should display them as text.
    * Texting/chat applications should display them as images.
    * Web pages should display some of them as text and others as images depending on whether they have the `Emoji_Presentation` property or not.

Basically, the EmojiOne library uses the presentation normally associated with texting/chat, not web pages. I believe that's what Twemoji does too.

Here's a page that displays emoji grouped by their default presentation style: http://unicode.org/emoji/charts/emoji-style.html

**Comment by caseyahenson** [comment]
The 3.0 libraries now consider greedy matching as an optional, user-defined metric for matching unicode data. The copyright symbol, among others, will only be matched when paired with a vs16 selector, or without vs16 when greedy matching is turned on.
